core/src/Kokkos_ScratchSpace.hpp

```diff
@@ -110,7 +110,7 @@ class ScratchMemorySpace {
     // Note: for team scratch m_offset is 0, since every
     // thread will get back the same shared pointer
     void* tmp           = m_iter + m_offset * size;
-    uintptr_t increment = size * m_multiplier;
+    uintptr_t increment = static_cast<uintptr_t>(size) * m_multiplier;
```

Comment:
Probably worth tested.

Reply:
So, is it a compiler warning/error you're fixing or rather some runtime issue?

Reply:
https://github.com/kokkos/kokkos/security/code-scanning/24

Reply:
I guess it's worth adding this `clang-tidy` check once you're done.

Reply:
Sure.  It looks like it was added in LLVM 13.

Reply:
@dalg24 Followup then ?

Reply:
What are you asking?

Reply:
Sure we probably should work towards enabling all the bugprone-* checks, but no I am not working on it.